web/src/components/core/Sidebar.test.jsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { screen, within, createEvent, fireEvent } from "@testing-library/react";
+import { plainRender, mockComponent, mockLayout } from "@/test-utils";
+import { Sidebar } from "@components/core";
+
+jest.mock("@components/layout/Layout", () => mockLayout());
+jest.mock("@components/core/About", () => mockComponent("About Mock"));
+jest.mock("@components/core/ChangeProductButton", () => mockComponent("ChangeProductButton Mock"));
+jest.mock("@components/core/LogsButton", () => mockComponent("LogsButton Mock"));
+jest.mock("@components/network/TargetIpsPopup", () => mockComponent("Host Ips Mock"));
+
+it("renders the sidebar initially hidden", async () => {
+  plainRender(<Sidebar />);
+  const nav = await screen.findByRole("navigation", { name: /options/i });
+  expect(nav).toHaveAttribute("data-state", "hidden");
+});
+
+it("renders a link for changing the sidebar to visible", async () => {
+  const { user } = plainRender(<Sidebar />);
+
+  const link = await screen.findByLabelText(/Open/i);
+  const nav = await screen.findByRole("navigation", { name: /options/i });
+
+  expect(nav).toHaveAttribute("data-state", "hidden");
+  await user.click(link);
+  expect(nav).toHaveAttribute("data-state", "visible");
+});
+
+// Test that opening the sidebar does not navigate to the initial route
+// This is achive by preventing the default link click behavior
+// Read https://testing-library.com/docs/dom-testing-library/api-events#fireevent and
+// https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented
+it("prevents the default event when the user click on the open link", async () => {
+  plainRender(<Sidebar />);
+  const link = await screen.findByLabelText(/Open/i);
+  const clickEvent = createEvent.click(link);
+  fireEvent(link, clickEvent);
+  expect(clickEvent.defaultPrevented).toBe(true);
```

Comment:
Sure, this is a quite particular test that must not be used as an example for other components. But I want to keep it because the intention is to avoid a _regression_ when opening the sidebar: I myself faced the problem of accidentally navigating from _Product Selection_ page to the _Overview_ when opening the sidebar.

Hopefully, this will change in the near future, after sanitizing the correct use of  `button` or `hiperlink`.

> In no particular order, below are the next UI changes I'd like to make
> 
>    * To fix the misuse of buttons/links
> 
>       https://ashleemboyer.com/blog/should-i-use-a-button-or-a-link

<sup>^^^ An excerpt from https://github.com/yast/d-installer/pull/391 description. ^^^</sup>